src/Extensions/AzureBlobPayloads/Interceptors/BasePayloadInterceptor.cs

```diff
@@ -0,0 +1,225 @@
+        {
+            await this.ResolveResponsePayloadsAsync(msg, ct);
+            return msg;
+        });
+
+        return new AsyncServerStreamingCall<TResponse>(
+            wrapped,
+            call.ResponseHeadersAsync,
+            call.GetStatus,
+            call.GetTrailers,
+            call.Dispose);
+    }
+
+    /// <summary>
+    /// Externalizes large payloads in request messages.
+    /// </summary>
+    /// <typeparam name="TRequest">The request type.</typeparam>
+    /// <param name="request">The request to process.</param>
+    /// <param name="cancellation">Cancellation token.</param>
+    /// <returns>A task representing the async operation.</returns>
+    protected abstract Task ExternalizeRequestPayloadsAsync<TRequest>(TRequest request, CancellationToken cancellation);
+
+    /// <summary>
+    /// Resolves payload tokens in response messages.
+    /// </summary>
+    /// <typeparam name="TResponse">The response type.</typeparam>
+    /// <param name="response">The response to process.</param>
+    /// <param name="cancellation">Cancellation token.</param>
+    /// <returns>A task representing the async operation.</returns>
+    protected abstract Task ResolveResponsePayloadsAsync<TResponse>(TResponse response, CancellationToken cancellation);
+
+    /// <summary>
+    /// Externalizes a payload if it exceeds the threshold.
+    /// </summary>
+    /// <param name="assign">Action to assign the externalized token.</param>
+    /// <param name="value">The value to potentially externalize.</param>
+    /// <param name="cancellation">Cancellation token.</param>
+    /// <returns>A task representing the async operation.</returns>
+    protected Task MaybeExternalizeAsync(Action<string?> assign, string? value, CancellationToken cancellation)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            return Task.CompletedTask;
+        }
+
+        int size = Encoding.UTF8.GetByteCount(value);
+        if (size < this.options.ExternalizeThresholdBytes)
+        {
+            return Task.CompletedTask;
+        }
+
+        return UploadAsync();
+
+        async Task UploadAsync()
+        {
+            string token = await this.payloadStore.UploadAsync(Encoding.UTF8.GetBytes(value), cancellation);
+            assign(token);
+        }
+    }
+
+    /// <summary>
+    /// Resolves a payload token if it's known to the store.
+    /// </summary>
+    /// <param name="assign">Action to assign the resolved value.</param>
+    /// <param name="value">The value to potentially resolve.</param>
+    /// <param name="cancellation">Cancellation token.</param>
+    /// <returns>A task representing the async operation.</returns>
+    protected async Task MaybeResolveAsync(Action<string?> assign, string? value, CancellationToken cancellation)
+    {
+        if (string.IsNullOrEmpty(value) || !this.payloadStore.IsKnownPayloadToken(value))
```

Comment:
Is `IsKnownPayloadToken` essentially being used as a versioning mechanism? It seems like it may cause problems if we try and extend it in the future. Or is the hope to use it to distinguish between different storage backends?